libcontainerd/remote/client.go

```diff
@@ -193,64 +111,36 @@ func (c *client) Restore(ctx context.Context, id string, attachStdio libcontaine
 		pid = int(t.Pid())
 	}
 
-	cntr.bundleDir = filepath.Join(c.stateDir, id)
-	cntr.ctr = ctr
-	cntr.task = t
-	// TODO(mlaventure): load execs
-
 	c.logger.WithFields(logrus.Fields{
 		"container": id,
 		"alive":     alive,
 		"pid":       pid,
 	}).Debug("restored container")
-
 	return alive, pid, nil
 }
 
-func (c *client) Create(ctx context.Context, id string, ociSpec *specs.Spec, runtimeOptions interface{}) error {
-	if ctr := c.getContainer(id); ctr != nil {
-		return errors.WithStack(errdefs.Conflict(errors.New("id already in use")))
-	}
-
-	bdir, err := prepareBundleDir(filepath.Join(c.stateDir, id), ociSpec)
-	if err != nil {
-		return errdefs.System(errors.Wrap(err, "prepare bundle dir failed"))
-	}
-
-	c.logger.WithField("bundle", bdir).WithField("root", ociSpec.Root.Path).Debug("bundle dir created")
-
-	cdCtr, err := c.client.NewContainer(ctx, id,
+func (c *client) Create(ctx context.Context, id string, ociSpec *specs.Spec, runtimeOptions interface{}) (containerd.Container, error) {
+	container, err := c.client.NewContainer(ctx, id,
 		containerd.WithSpec(ociSpec),
-		containerd.WithRuntime(runtimeName, runtimeOptions))
+		containerd.WithRuntime(runtimeName, runtimeOptions),
+		withBundle(filepath.Join(c.stateDir, id), ociSpec),
+	)
 	if err != nil {
-		return wrapError(err)
+		return nil, wrapError(err)
 	}
-
-	c.Lock()
-	c.containers[id] = &container{
-		bundleDir: bdir,
-		ctr:       cdCtr,
-	}
-	c.Unlock()
-
-	return nil
+	return container, nil
 }
 
 // Start create and start a task for the specified containerd id
 func (c *client) Start(ctx context.Context, id, checkpointDir string, withStdin bool, attachStdio libcontainerdtypes.StdioCallback) (int, error) {
-	ctr := c.getContainer(id)
-	if ctr == nil {
-		return -1, errors.WithStack(errdefs.NotFound(errors.New("no such container")))
-	}
-	if t := ctr.getTask(); t != nil {
-		return -1, errors.WithStack(errdefs.Conflict(errors.New("container already started")))
+	ctr, err := c.getContainer(ctx, id)
+	if err != nil {
+		return -1, wrapError(err)
```

Comment:
I see `wrapError` only handles "notFound" errors; is it possible other type of errors are returned?

Reply:
not in this pr

Reply:
Was asking because I saw before this change some other type of errors were returned; just to be sure we don't regress.

I'm ok with adding more types in a follow up